lapce-data/src/db.rs

```diff
@@ -518,7 +560,30 @@ impl LapceDb {
         let workspace = (*data.workspace).clone();
         let workspace_info = data.workspace_info();
 
+        // Buffer for auto save on quit
+        let main_split = &data.main_split;
+
         self.insert_workspace(&workspace, &workspace_info)?;
+        self.insert_unsaved_buffer(main_split)?;
+
+        Ok(())
+    }
+
+    fn insert_unsaved_buffer(&self, main_split: &LapceMainSplitData) -> Result<()> {
+        let sled_db = self.get_db()?;
+
+        for (path, doc) in &main_split.open_docs {
+            if !doc.buffer().is_pristine() && doc.content().is_file() {
+                let path_str = path.to_str().unwrap();
+                let buf_text = doc.buffer().text().to_string();
+                sled_db.insert(
```

Comment:
I think we need to insert all unsaved buffers as one key, so that it can override it when the file is saved.

Reply:
What would be the advantage compared to what is done right now?
it inserts a new key for each unsaved file, and on startup, the data is popped and then removed (see `lapce-data/src/db.rs:464`)